### ISSUE TITLE
TCP ClientProxy strange behavior when returning value from catchError in observable

### ISSUE THREAD

**Issue by dvdalilue** [nestjs/nest#4049]
## Bug Report
I'm having an issue with my microservices communication.
## Current behavior

Currently, I have two microservices, a resource provider and another one is a consumer. Also, the provider is connected to a DB through a gRPC client (ClientGrpc). I've defined a controller in my provider which recognizes one message using the decorator `MessagePattern` and returns a `boolean`. The controller uses a service where the logic is implemented and the DB access performed. The problem is when the consumer sends a message and gets an unexpected response in some cases. I'm just checking an item existence, if exist return `true`, otherwise returns `false` and any other case, raise an exception. See the following code. 

## Input Code

`provider.controller.ts`:
```ts
    @MessagePattern({ cmd: 'check' })
    check(params: any): Observable<boolean> {
        return this.service.check(params.id);
    }
```

`service.ts`:
```ts
    check(id: string): Observable<boolean> {
        return this.gRpcClientDB.getItem({ id }).pipe(
            first(),
            map((_) => { return true; }),
            catchError((error) => {
                if (error.code == MyExceptionStatus.NOT_FOUND) {
                    return of(false);
                }
                raise_error(error); // Error filter/hadler
            })
        );
    }
```

`consumer.controller.ts`:
```ts
    @Get()
    checkItem(): Observable<boolean> {
        const pattern = { cmd: 'check' };
        const data = { id: 'my_id' };
        return this.tcpProviderClient.send<boolean>(pattern, data).pipe(
            first(),
            catchError((error) => {
                raise_error(error);
            })
        );
    }
```
But I'm not getting the not-found case correctly, the consumer receives nothing in the observable unless I use the following code in `service.ts`.

`service.v2.ts`:
```ts
    check(id: string): Observable<boolean> {
        return this.gRpcClientDB.getItem({ id }).pipe(
            first(),
            map((_) => { return true; }),
            catchError((error) => {
                if (error.code == MyExceptionStatus.NOT_FOUND) {
                    return of(false, false); // or of(false, true); which is strange
                }
                raise_error(error); // Error filter/hadler
            })
        );
    }
```

## Expected behavior

I expect to get true if an item exists (this happens), false otherwise (the strange behavior).

## Possible Solution

I'm not quite sure.

## Environment

<pre><code>
Nest version: 6.14.1

 
For Tooling issues:
- Node version: 12.14.1  
- Platform: Linux Ubuntu 18.04.1 x86_64  

Others:

</code></pre>

**Comment by kamilmysliwiec** [comment]
Please, provide a minimal repository which reproduces your issue.

**Comment by dvdalilue** [comment]
No problem, [here](https://github.com/dvdalilue/nestjs-microservices) is a repo with the issue. I just realized the problem is with boolean (maybe another type) but with numbers works fine.

**Comment by kamilmysliwiec** [comment]
Thanks for sharing. Fixed in 6.11.7
